SonarScanner.Shim/PropertiesFileGenerator.cs

```diff
@@ -257,9 +256,23 @@ private static ProjectInfoAnalysisResult ProcessProjectInfoFiles(IEnumerable<Pro
 
                 result.Projects.Add(projectInfo, status);
             }
+            result.SharedFiles.RemoveWhere(s => IsPartOfAProject(projects, s));
+
             return result;
         }
 
+        private static bool IsPartOfAProject(IEnumerable<ProjectInfo> projects, string file)
+        {
+            foreach (ProjectInfo projectInfo in projects)
```

Comment:
Could be rewritten using Linq. 
`return projects.Any(projectInfo =>IsInFolder(file, projectInfo.GetProjectDirectory()));`